src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -117,7 +117,13 @@ ZeroInferRequest::ZeroInferRequest(const std::shared_ptr<ZeroInitStructsHolder>&
                                               executorInputDescriptors.at(shapeBufferName),
                                               shapeBufferName);
 
-            auto allocator = zeroMemory::HostMemAllocator(backendPtr);
+            ov::Allocator allocator;
+            if (properties.flags & ZE_DEVICE_PROPERTY_FLAG_INTEGRATED) {
+                allocator = zeroMemory::HostMemAllocator(backendPtr, ZE_HOST_MEM_ALLOC_FLAG_BIAS_WRITE_COMBINED);
+            } else {
+                allocator = zeroMemory::HostMemAllocator(backendPtr);
+            }
```

Comment:
```suggestion
            ov::Allocator allocator = [&] {
                if (properties.flags & ZE_DEVICE_PROPERTY_FLAG_INTEGRATED) {
                    return zeroMemory::HostMemAllocator(backendPtr, ZE_HOST_MEM_ALLOC_FLAG_BIAS_WRITE_COMBINED);
                }
                return zeroMemory::HostMemAllocator(backendPtr);
            }();
```

A little shorter and initializes allocator.

Reply:
I kept the old implementation here with the thought of reusing the allocator for both inputs and shape input tensors.